src/Client/OAuth2Client.php

```diff
@@ -112,6 +113,23 @@ public function getAccessToken(array $options = [])
         );
     }
 
+    /**
+     * Refresh the given AccessToken.
+     *
+     * @param array $options Additional options that should be passed to the getAccessToken() of the underlying provider
+     *
+     * @return AccessToken|\League\OAuth2\Client\Token\AccessTokenInterface
+     *
+     * @throws IdentityProviderException If token cannot be fetched
+     */
+    public function refreshAccessToken(AccessTokenInterface $accessToken, array $options = [])
```

Comment:
Something just occurred to me. Let's think of the flow:

1) After the OAuth flow, I call `getAccessToken()` and get back the `AccessToken` object.
2) I decide to store this in the database - on an `User.accessToken` property
3) Now... I no longer have the `AccessToken` object, I only have the access token string. So, I can't use `$accessToken->hasExpired()` (like the new docs show) to see if the token has expired. But more importantly, I can't call this method because I don't have the object anymore. 

WDYT? Am I missing something? Would you recommend a different strategy for storing the access token object? Or should this method's first argument be `string $refreshToken` and then we document how you should store the refresh token string in (e.g.) the database?

Reply:
That's a good point. The alternative (and what I'm currently using) is to store the whole AccessToken (ie. token string, refresh token and expiry), so that you can check if it's expired and refresh if required.

However, just storing the refresh token and refreshing everytime you want to use it is a more simple flow.

I think making this method take the `string $refreshToken` is the most flexible, it allows both flows.